### ISSUE TITLE
Client Branding/Marketing Sites: myproject.org

### ISSUE THREAD

**Issue by cleverington** [elmsln/openulmus.org#32]
## Observation
OpenUlmus will provide areas for each community to host their _Courses_, but does it provide areas to host their Branding sites?

For example, (as a Customer) if I get `courses.myproject.org` setup, can I also get `myproject.org` to host a marketing/branding site?

## Tiered Pricing

I'm going to recommend this _not_ be a complimentary service, as Group Owners may create their marketing/branding site and never actually create any _paid_ Courses to assist in supporting the collective.  One or two would not be a problem, but if every community throws out ten groups doing the same thing, sustainability will quickly become a problem.

Instead, I would recommend having it as an option only for those on the _Advanced_ package, or above (see https://github.com/elmsln/openulmus.org/wiki/6.-Budget-and-Sustainability ).  (eg: those paying $500.00 a month, or more.)

## Automation

What tasks would be available for pointing a domain at my area on openulmus?

Are Customers going to have SSH access or will the OpenUlmus Team be setting up these sites for them?

Can we leverage something like GitHub Webhooks to auto-deploy branding/marketing sites so Clients/Customers _no not_ have access?

**Comment by gusaus** [comment]
Agreed that we should not be hosting branding sites for free. I'm not sure if/how it could fit in with the current ELMS:LN workflow, but I 'think' the next release of Aegir (AegirNG) will provide us with a way to auto-deploy GravCMS based branding/marketing sites.

Some of the scenarios listed in the AegirNG Architecture (https://gitlab.com/aegir/aegir/wikis/architecture) are close to how we're already using Aegir for OpenProducer https://github.com/OpenProducer/administration/issues/2. With a good deal of potential overlap here, we should see if @ergonlogic @jonpugh and of course @btopro have any thoughts on this issue. 

Everybody is also reachable via Open Collective slack.

**Comment by btopro** [comment]
we've discussed adding things like this to elms:ln in the form of program hub / program level marketing. This sounds more in scope of AEgir but whatever gets the job done. This could also be some minor custom automation to request the top level site if needed

**Comment by mmilutinovic013** [comment]
I like the sustainability model for us hosting the site for these groups. This allows the group the option to let us take care of hosting for a cost - but does not burden us by hosting more domains that may not even be desired by the client base. We need ways to promote monthly subscription contributions to ensure stability -- so I agree with @cleverington 's comments above on the model. 

On the technical side -- I think this is what @btopro and I are working on this coming Tuesday. So we should have the capability by then (or soon thereafter).

**Comment by gusaus** [comment]
Here's some more info from some of the Aegir leads regarding support of Grav (and other non-Drupal platforms) - https://github.com/elmsln/openulmus.org/issues/32#issuecomment-345476831
![screenshot_20171121-193332](https://user-images.githubusercontent.com/113996/33105747-a47d53d0-cee3-11e7-9e28-99821cd07acf.png)

Knowing that, possibly Aegir as a way to spin up stand alone sites (including Grav and Open Social based) could be incorporated into our Open Ulmus toolkit? 

It hurts my head just thinking about it, but we could loop @ergonlogic and @jonpugh into the scoping discussion? https://github.com/elmsln/openulmus.org/issues/1

**Comment by colans** [comment]
> We need ways to promote monthly subscription contributions to ensure stability

Not sure if you folks have thought about this or already have another solution, but I've been working on the e-commerce part of Aegir (allowing users to get their own sites and pay for them with recurring billing) via [Recurly Aegir](https://www.drupal.org/project/recurly_aegir).  When there's help, time and/or funding, this will eventually morph into [Aegir Site Subscriptions](https://www.drupal.org/sandbox/tomcox/2816099), which will support other payment back-ends, not just Recurly.

As the framework is being written in a generic way, it shouldn't be too much trouble to support different versions of Aegir (3, 4, or 5).  It's only the format of the Web service calls that would need to change, if at all.

**Comment by gusaus** [comment]
@colans It would be great to further explore if/how Aegir could fit into the Open Ulmus flow. As the latter is built on ELMS:LN, we'd definitely need guidance from @btopro regarding any sort of network integration (excuse me for not knowing what I'm talking about 😕)    https://elmsln.readthedocs.io/en/latest/ia/System/

Based on the clarity provided by you, @ergonlogic & @jonpugh I'm confident we have the collective resources to move forward with what I guess could be a joint, white label SaaS offering for Open Producer. We could move that discussion to related issues and have a follow discussion in Drupal or OpenCollective Slack. https://github.com/OpenProducer/administration/issues/2 https://github.com/OpenProducer/openproducer/issues/1

**Comment by gusaus** [comment]
Quick followup to @colans https://github.com/elmsln/openulmus.org/issues/32#issuecomment-346372025 and subsequent discussions in chat. 

>   Not sure if you folks have thought about this or already have another solution, but I've been working on the e-commerce part of Aegir (allowing users to get their own sites and pay for them with recurring billing) via Recurly Aegir. When there's help, time and/or funding, this will eventually morph into Aegir Site Subscriptions, which will support other payment back-ends, not just Recurly.

Could https://www.drupal.org/sandbox/tomcox/2816099 eventually provide some way to integrate Open Collective? https://github.com/opencollective/opencollective/wiki#devs

As it will 'soon' provide a mechanism to spin up stand alone sites (including Grav and Open Social based) could we incorporate Aegir into our Open Ulmus white label PaaS offering on AWS?

cc @djfusco as evaluating these types of options may be in scope with https://github.com/elmsln/openulmus.org/issues/2#issuecomment-346379850

**Comment by djfusco** [comment]
@gusaus this may help explain an option in spinning up ELMSLN fairly quickly into AWS.  Of course, there's some knowledge that needs to be had re: AWS provisioning and DNS, but I automated the rest from Vagrant.  Took like 15 min to run.  Lemme know if you want to discuss.  https://github.com/djfusco/Vagrant-AWS

**Comment by gusaus** [comment]
@djfusco Aside from @btopro, you're probably going to be the primary person evaluating these options and helping incorporate into the Open Ulmus dev flow. I'm getting confused just pondering whether what you're mentioning is an option for spinning up Grav sites. 

Join #elmsln in https://slack.opencollective.com/ if you can.

**Comment by gusaus** [comment]
Based on recent conversations in Slack, I'm wondering if it wouldn't be a bad idea to include Aegir in our dev-flow so we could spin up, and offer subscriptions for, platforms that may not yet be included in Open Ulmus. For instance we could provide partners/clients Open Social or some other flavor of Drupal. We can also provide something like GravCMS as a stand-alone marketing site. 

This could provide us with more flexibility and additional revenue streams. 

Thoughts from the group?

**Comment by cleverington** [comment]
@mmilutinovic1313 As a forewarning, once I do this, you'll need to update the remotes on both your local and server copy of git.

Running the command `git remote -v` **now** will give you:
```
origin	git@github.com:elmsln/openulmus.org.git (fetch)
origin	git@github.com:elmsln/openulmus.org.git (push)
```

Once I make these changes, you'll need to change where it is pointing:

`git remote set-url origin git@github.com:openulmus/openulmus.org.git`

I'll let you know once the change needs to be made.

**Comment by gusaus** [comment]
Do we need to finish up https://github.com/openulmus/openulmus.org/issues/34 before we can spin up Open Ulmus for partners? https://github.com/openlearninglabs/administration/issues/6#issue-277957596 

@btopro Is Grav part of ELMS:LN core? 

We would also like to provide the completed setup ([Grav skeleton](https://learn.getgrav.org/advanced/grav-development#grav-skeletons)?) as a stand alone platform on our Aegir setup for Open Learning Labs and Open Producer https://github.com/openulmus/openulmus.org/issues/32#issuecomment-346457797

If @jonpugh already has this working, possibly we can work with him to update our Aegir setup to Provision 4  https://twitter.com/jonpugh/status/961704837502111744

**Comment by btopro** [comment]
@gusaus GravCMS is in there, it has not been utilized and would need some work tho for the purposes that you desire I mean it can just be thrown on any sever; it's more the standing up promo-sites OOTB / provisioning automatically (those blurry areas where AEgir and ELMS:LN overlap architecturally). I'd just have someone setup a gravsite and not worry about that part of elms:ln currently. We're core'ing grav more for future portfolio site capabilities (and mini-sites as a side-benefit)

**Comment by gusaus** [comment]
Thanks @btopro - based on what you're saying, I think setting up an Aegir based dev & production environment would be a best way to create some skeletons that can be committed back to ELMS:LN and Grav. Maybe that could overlap with what @paulhibbitts has been doing with the Grav based Course Hub https://github.com/hibbitts-design/grav-skeleton-course-hub

**Comment by gusaus** [comment]
Circling back to this issue to see what options we have for developing and spinning up Grav sites since https://github.com/openulmus/openulmus.org/issues/32#issuecomment-369088765

Not sure if DDEV as discussed in https://github.com/openulmus/openulmus.org/issues/48#issuecomment-396415511 could be incorported into our dev --> AWS flow.
